consoles/sshVirtsh.pm

```diff
@@ -111,27 +111,34 @@ sub _init_xml {
     $elem->appendTextNode($self->vmm_type);
     $os->appendChild($elem);
 
-    if ($self->vmm_family eq 'xen') {
-        if ($self->vmm_type eq 'hvm') {
-            my $features = $doc->createElement('features');
-            $root->appendChild($features);
-
-            $elem = $doc->createElement('acpi');
-            $features->appendChild($elem);
-            $elem = $doc->createElement('apic');
-            $features->appendChild($elem);
+    if (($self->vmm_family eq 'xen' and $self->vmm_type eq 'hvm') or get_var('UEFI')) {
+        my $features = $doc->createElement('features');
+        $root->appendChild($features);
+        $elem = $doc->createElement('acpi');
+        $features->appendChild($elem);
+        $elem = $doc->createElement('apic');
+        $features->appendChild($elem);
+        if ($self->vmm_family eq 'xen' and $self->vmm_type eq 'hvm') {
             $elem = $doc->createElement('pae');
             $features->appendChild($elem);
         }
-        elsif ($self->vmm_type eq 'linux') {
-            $elem = $doc->createElement('kernel');
-            $elem->appendTextNode('/usr/lib/grub2/x86_64-xen/grub.xen');
-            $os->appendChild($elem);
-        }
+    }
+
+    if ($self->vmm_family eq 'xen' and $self->vmm_type eq 'linux') {
+        $elem = $doc->createElement('kernel');
+        $elem->appendTextNode('/usr/lib/grub2/x86_64-xen/grub.xen');
+        $os->appendChild($elem);
+    }
+
+    if (get_var('UEFI')) {
+        $elem = $doc->createElement('loader');
+        $elem->appendTextNode('/usr/share/qemu/ovmf-x86_64-ms-code.bin');
```

Comment:
The qemu backend has actually a search path - using a different path than you. Doing the same with svirt is a bit harder, but still the qemu backend has proven you can't just hardcode SUSE's path as generic.